cloud/azure/azure_rm_storageblob.py

```diff
@@ -0,0 +1,603 @@
+            - Source file path. Use with state 'present' to upload a blob.
+        aliases:
+            - source
+        required: false
+        default: null
+    state:
+        description:
+            - Assert the state of a container or blob.
+            - Use state 'absent' with a container value only to delete a container. Include a blob value to remove
+              a specific blob. A container will not be deleted, if it contains blobs. Use the force option to override,
+              deleting the container and all associated blobs.
+            - Use state 'present' to create or update a container and upload or download a blob. If the container
+              does not exist, it will be created. If it exists, it will be updated with configuration options. Provide
+              a blob name and either src or dest to upload or download. Provide a src path to upload and a dest path
+              to download. If a blob (uploading) or a file (downloading) already exists, it will not be overwritten
+              unless the force parameter is true.
+        default: present
+        required: false
+        choices:
+            - absent
+            - present
+    public_access:
+        description:
+            - Determine a container's level of public access. By default containers are private. Can only be set at
+              time of container creation.
+        required: false
+        default: null
+        choices:
+            - container
+            - blob
+    tags:
+        description:
+            - "Dictionary of string:string pairs to assign as metadata to the object. Metadata tags on the object
+              will be updated with any provided values. To remove tags use the purge_tags option."
+        required: false
+        default: null
+    purge_tags:
+        description:
+            - Use to remove tags from an object. Any tags not found in the tags parameter will be removed from
+              the object's metadata.
+        required: false
+        default: false
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Chris Houseknecht (@chouseknecht)"
+    - "Matt Davis (@nitzmahone)"
+
+'''
+
+EXAMPLES = '''
+- name: Remove container foo
+  azure_rm_storageblob:
+    resource_group: testing
+    storage_account_name: clh0002
+    container: foo
+    state: absent
+
+- name: Create container foo and upload a file
+  azure_rm_storageblob:
+    resource_group: Testing
+    storage_account_name: clh0002
+    container: foo
+    blob: graylog.png
+    src: ./files/graylog.png
+    public_access: container
+    content_type: 'application/image'
+
+- name: Download the file
+  azure_rm_storageblob:
+    resource_group: Testing
+    storage_account_name: clh0002
+    container: foo
+    blob: graylog.png
+    dest: ~/tmp/images/graylog.png
+'''
+
+RETURN = '''
+changed:
+    description: Whether or not the object was changed.
+    returned: always
+    type: bool
+    sample: True
+actions:
```

Comment:
this should not be a default, either returned with `diff` or output through the debug mode